Android13

theimpulson opened the pull request:
## Summary

This PR adds initial support for Android 13

## Issue(s)

https://github.com/seedvault-app/seedvault/issues/433

## Screenshot

None are available due to several changes. Please test manually.

## Test

To test this MR locally, use the following commands to fetch the changes:

```shell
git clone git@github.com:seedvault-app/seedvault.git && cd ./seedvault
git fetch upstream pull/434/head && git checkout FETCH_HEAD
```

theimpulson committed:
gradle: Bump to latest stable AGP and gradle versions

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
gradle: Bump kotlin and dependencies version for Android 13

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
storage: Address experimental time API errors

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
storage: Switch to inWholeMilliseconds property

W: 'toLongMilliseconds(): Long' is deprecated. Use inWholeMilliseconds property instead.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
{seedvault|storage}: Add explicit dependencies for kotlin coroutines

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
gradle: Enable java 11 language features

Also unify the style while at it

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
gradle: Bump SDK versions to 33

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
gradle: Upgrade robolectric and add explicit dependency on hamcrest

Current hamcrest dependency was coming from robolectric but upgrading it
removed it.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

grote commented on app/src/test/java/com/stevesoltys/seedvault/restore/install/ApkBackupRestoreTest.kt:
```diff
@@ -133,7 +133,9 @@ internal class ApkBackupRestoreTest : TransportTest() {
-        every { pm.getPackageArchiveInfo(capture(apkPath), any()) } returns packageInfo
+        every {
+            pm.getPackageArchiveInfo(capture(apkPath), PackageManager.PackageInfoFlags.of(any()))
+        } returns packageInfo
```
How does it work to switch the tests to the new method, but not the actual code that is being tested?

grote commented on app/build.gradle:
```diff
@@ -138,10 +138,11 @@ dependencies {
     testImplementation rootProject.ext.aosp_libs
     testImplementation 'androidx.test.ext:junit:1.1.3'
-    testImplementation('org.robolectric:robolectric:4.3.1') { // 4.4 has issue with non-idle Looper
+    testImplementation('org.robolectric:robolectric:4.8.1') { // 4.4 has issue with non-idle Looper
```
If `4.8.1` works, we can remove the comment about `4.4`.

theimpulson committed:
settings: AppListRetriever: Switch to lowercase() method

W: 'toLowerCase(Locale): String' is deprecated. Use lowercase() instead.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

grote commented on app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt:
```diff
@@ -134,6 +137,14 @@ internal class PackageService(
     }
 
+    private fun PackageManager.getInstalledPackagesCompat(flags: Int): MutableList<PackageInfo> {
+        return if (SDK_INT >= TIRAMISU) {
```
Usually, we use the integer here, because who can remember all those desert names?

grote replied:
Btw., If there's no more issue with Roboelectric, I also don't mind bumping minSdk to 33.

grote replied:
Hmpf, I am getting `android.content.pm.PackageParser$PackageParserException: seedvault/app/build/intermediates/apk_for_local_test/debugUnitTest/apk-for-local-test.ap_ (at Binary XML file line #7): Requires newer sdk version #33 (current version is #29)
` with minSdk 33.

and if settings tests to use 33, I get `API level 33 is not available`. So maybe just leave the deprecated method until repoelectric can handle 33?

theimpulson replied:
Killed as discussed

grote commented on app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt:
```diff
@@ -134,6 +137,14 @@ internal class PackageService(
         }
     }
 
+    private fun PackageManager.getInstalledPackagesCompat(flags: Int): MutableList<PackageInfo> {
```
Does the list need to be mutable?

theimpulson replied:
Killed as discussed

grote committed:
Mock ApplicationInfo

as it now calls the native SystemClock.uptimeMillis() method which is hard to mock

grote reviewed (commented):
Reminder: We still need to update `/contactsbackup/libs/com.android.vcard.jar` and maybe add this already to some porting docs, so we won't forget this in the future.

theimpulson committed:
restore: install: Cast any() to Int to avoid oevrload errors

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>
